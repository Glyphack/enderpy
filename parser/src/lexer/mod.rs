mod diagnostics;
pub mod fstring;
pub mod lexer;
