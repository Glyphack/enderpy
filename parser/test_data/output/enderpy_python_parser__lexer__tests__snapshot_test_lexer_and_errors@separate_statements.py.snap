---
source: parser/src/lexer/mod.rs
description: "# Test case to check that we return correct error when two python statements are on the same line\n\ndef foo():\n    x = 1; y = 2\n    return x + y\n\ndef bar():\n    # TODO(parser): enable after error handling\n    # x = 1 y = 2\n    return x + y\n\n# TODO(parser): enable after error handling\n# a = 1  b = 2\n"
input_file: parser/test_data/inputs/separate_statements.py
---
0,97: Comment   (Str("# Test case to check that we return correct error when two python statements are on the same line"))
97,98: NewLine   (None)
98,99: NewLine   (None)
99,102: Def   (None)
103,106: Identifier   (Str("foo"))
106,107: (   (None)
107,108: )   (None)
108,109: :   (None)
109,110: NewLine   (None)
110,114: Indent   (Indent(1))
114,115: Identifier   (Str("x"))
116,117: =   (None)
118,119: Integer   (Number("1"))
119,120: ;   (None)
121,122: Identifier   (Str("y"))
123,124: =   (None)
125,126: Integer   (Number("2"))
126,127: NewLine   (None)
131,137: Return   (None)
138,139: Identifier   (Str("x"))
140,141: +   (None)
142,143: Identifier   (Str("y"))
143,144: NewLine   (None)
144,145: NewLine   (None)
145,145: Dedent   (Indent(1))
145,148: Def   (None)
149,152: Identifier   (Str("bar"))
152,153: (   (None)
153,154: )   (None)
154,155: :   (None)
155,156: NewLine   (None)
156,160: Indent   (Indent(1))
160,203: Comment   (Str("# TODO(parser): enable after error handling"))
203,204: NewLine   (None)
208,221: Comment   (Str("# x = 1 y = 2"))
221,222: NewLine   (None)
226,232: Return   (None)
233,234: Identifier   (Str("x"))
235,236: +   (None)
237,238: Identifier   (Str("y"))
238,239: NewLine   (None)
239,240: NewLine   (None)
240,240: Dedent   (Indent(1))
240,283: Comment   (Str("# TODO(parser): enable after error handling"))
283,284: NewLine   (None)
284,298: Comment   (Str("# a = 1  b = 2"))
298,299: NewLine   (None)
